Implement `pulp version`

hdgarrood opened the pull request:
This gets us most of the way towards #57. :)

By the way, I think I worked out why the Windows builds are failing: https://github.com/natefaubion/purescript-psa/issues/13

hdgarrood committed:
Implement `pulp version`

hdgarrood committed:
Remove unused imports in Pulp.Login

hdgarrood committed:
Don't print errors when we're about to throw them

hdgarrood committed:
Fix warnings

bodil commented on src/Main.purs:
```diff
@@ -198,7 +199,9 @@ commands = [
-  Args.command "login" "Obtain and store a token for uploading packages to Pursuit." Nothing Login.action []
+  Args.command "login" "Obtain and store a token for uploading packages to Pursuit." Nothing Login.action [],
+  Args.commandWithArgs "version" "Bump and tag a new version in preparation for release." Nothing BumpVersion.action []
+    [Args.argument "bump" Type.versionBump "The type of bump to perform" false]
```
Could we change this help message to show a list of the available bump types instead?

hdgarrood replied:
Ah yes, that reminds me, the help is kind of useless for pulp version right now.

hdgarrood committed:
Attempt to fix tests

hdgarrood committed:
Better help for 'pulp version'

hdgarrood committed:
Fix documentation comments for Argument type

hdgarrood committed:
Only print command opts if there is at least 1

hdgarrood committed:
Fix `pulp version --help`

hdgarrood committed:
Better help display for commands with arguments

hdgarrood committed:
Remove version field from generated bower.json

'pulp version' doesn't use 'bower version' (because it's broken), and
the version field in bower.json is deprecated according to the spec
anyway, so stop adding version fields to bower.json files generated by
'pulp init'.

hdgarrood committed:
Stick to purescript 0.8.2 for now
